* TODO LaTeX Template [6/6]
  - [X] Fonts and titles
    + [X] Titles should be helvetica
    + [X] Space between title and text should be reduced
    + [X] Main titles should be left aligned
    + [X] Main title fonts need changing
  - [X] Header and footer
    + [X] Header should be left side for even page and right side for
      odd.
    + [X] Same with footer
    + [X] Left header should be author
    + [X] Right side should be Manuscript name
    + [X] Header and footer font should be helvetica and bold
    + [X] Headers should also be italic
    + [X] Footers should be page number
    + [X] More spacing for header and line
  - [X] Bibliography
    + [X] Cite by author and date in round brackets
    + [X] Choose a style option
  - [X] Try multifile
  - [X] Try refs (may need to use reftex)
  - [X] Add a spell checker to emacs

* TODO Proposal [3/10]
  - [X] Read refs
  - [ ] Add content from Conors notes
  - [X] Find images for Background
  - [X] Add refs to aims and images
  - [ ] Outline the Approach
  - [ ] Make a diagram/schematic of the model
  - [ ]	Make a Gantt chart / maybe just a rough one on paper
  - [ ] Write out equations for models in rst, markdown, or orgmode
  - [ ] Add detail to the approach
  - [ ] Add detail to the background
** Aims
  - [ ] HYPOTHESES [0/2]
    - [ ] QFA experiments are affected by competition for nutreints
      and/or signalling.
    - [ ] Reanalysis of telomere(?) data accounting for these effects
      will improve the reliability of predicted interactions and could
      provide evidence of new interactions.
  - [ ] What could the results of the project be used for?
    - [ ] To better understand systematic effects on population growth
      in agar assays.
    - [ ] To design assays to minimize competition and signalling
      effects
    - [ ] To account for these effects in the analysis of the data
      - [ ] Create a package(s) to do this for future experiments
      - [ ] Reanalyse some existing data - Specifically telomere?
	- [ ] The potential applications of QFA are broad and the
          approach could be used to study interactions with any viable
          query gene in bacteria or (what do we call the group of
          single celled eukaryotic organisms for which this is
          appropriate?) in high-throughput or smaller scale manual
          experiments.
** Objectives [8/9]
  - [X] Should be bullet point format.
  - [ ] Should be S.M.A.R.T. - I think they are.
  - [X] Develop a model of competition for nutrients and/or signalling
    between cultures on a QFA agar gel.
  - [X] Reanalyse telomere data from (ref), comparing the
    competition and independence models (a special case of the
    competition model), to determine the effect of competition on
    the experiments.
  - [X] Compare competition effects in different experimental
    designs in order to design experiments which minimze competition
    effects.
    - For instance, compare spot and agar geometries and initial nutrient
      densities.
  - [X] Develop a model of competition in terms of parameters of the
    logistic model of population growth in oder to compare fitness
    measurements to previous studies.
  - [X] Reanalyse telomere data with using the modified logistic
    model in order to determine the effects on gene interaction.
  - [X] Package the model in SBML. Conform to minimum information
    standards and publish in the biomodels database.
  - [X] Document and package other analysis tools in python in order
    to distribute.
** Background [0/4]
  - [ ] A comparison of methods for screening strain fitness
    - [ ] SGA
    - [ ] QFA
      - [ ] Lower densities possible so capture more of the growth
        curve
      - [ ] More than one colony per spot so will average out lag and
        stochastic effects/variance.
	- [ ] Possibility of competition within a spot.x
    - [ ] liquid cultures
    - [ ] Also explain why we want to model competition in order to
      get data from more of the growth curve
      - [ ] Batch variance (ref Bary2010) - want to fit as many plates
        into a batch as possible which limits the frequency of
        observations we can make.
  - [ ] Evidence of competition and signalling
    - [ ] Diffusion in agar
    - [ ] Types of signalling in yeast
      - [ ] ethanol
    - [ ] Evidence of competition in SGA and QFA
      - [ ] How large is the predicted effect
  - [ ] Modelling approaches
    - [ ] Fishers multiplicative model of genetic interactions (independence)
    - [ ] Continuity equation and diffusion equation
    - [ ] Mass action kinetics
    - [ ] Logistic model
    - [ ] Bayesian inference and heirarciacal bayesian models
      - [ ] With an explanation of the advantage of Bayesian approach
	- [ ] Over normalization in Baryshnikova 2010
	- [ ] Between Addinall2011 and Heydari2016
  - [ ] Inference of genetic interactions from fitness screening and telomeres
** Approach
   - Model competition for nutrients
     - Create a small set of artifical data (perhaps for nine spots at
       first) from the model and analyse this to see how long the
       analysis takes and how well we can recover the underlying
       parameters.
     - We can also write tests for the code at this stage which should
       scale to larger simulations.
     - Extend this to study experimental data.
       - A larger array (384 spots)
       - Fit competition model to growth curve data.
       - Compare with independent model (a special case of the
         competition model.)
     - Revise the model if necessary or add signalling factor.
     - Use in a reanalysis of (telomere?) data to discover genetic
       interactions.
   - [ ] Statistical methods
     - [ ] Least Squares
     - [ ] Heirarchical Bayesian model.
   - [ ] Computational tools
     - [ ] scipy for least squares
     - [ ] pyMC vs pySTAN for Bayesian inference. For pySTAN we would
       have to rewrite the ode model in Stan but it could be faster.
   - [ ] PACKAGING, SBML and MIRIAM


  - [ ] Develop a model of competition for nutrients and/or signalling
    between cultures on a QFA agar gel.
    - Nutrient dependent growth
      - Use deterministic mass action kinetics (already have model).
    - Singal dependent growth
      - Inhibiting
      - Rate of colony growth in mass action kinetic model of nutrient
        dependent growth reduces linearly up to some critical signal
        concentration where growth stops.
	- Here a yeast culture is inhibiting its own growth. Need to
          investigate in the literature what are the mechanisms of
          signalling and inhibition so that we can develop a
          physically accurate model as this is just my first guess.
    - Diffusion of nutrients and signal
      - Split the agar into a grid and model diffusion rate as
        lineraly dependent on the difference in the ammount of
        nurients/signal between spot locations.
      - This is a very simplified spatially-discretised 2d model of
        diffusion. (Similar to that used in imaging but with sinks.)
	- Is it accurate?
	- Could we use a finer grid (perhaps hexagonal)?
	- Is agar height important (3d grid)?
	- Are models already available for diffusion which we could use?
    - The independent model is just a special case of the competition
      and signalling model with diffusion constants set to zero (If
      discretized in a finer grid we would have to set boundaries at
      the edge of each location).
  - [ ] Test the model by reanalysing telomere data from (ref),
    comparing the competition and independence models (a special case
    of the competition model), to determine the effect of competition
    on the experiments.
    - Data points are obtained automatically by Colonyzer.
      - Colonyzer can fit a logistic growth model using what method?
      - Heydari et al. 2016 use a Bayesian heirarchical approach to
        simultaneously fit population dynamics (logistic growth
        model) and genetic interaction.
    - Fitting model to data.
      - If the model is written in SBML, how does this interface with python?
      - Soliving the model.
	- Constrain parameters with realistic values which are to
          be determined/guessed or discovered in the literature.
      - We will first fit our model for using a least squares method.
	- odeint - integrates a system of differential equations.
      - Bayesian inference
	- Not sure how we go about this yet.
	- But we would use pyMC or pySTAN. For pySTAN we would
	have to rewrite the ode model in Stan but it could be faster.
      - This is expected to take a long time so we need to find
        something else to do in the meantime
	- Write up sections of the research in a report or in a blog
	- Analyse past results and develop the next model while the
          current one is running.
	  - Will a linear dependence of development limit this sort of
            activity.
	- Optimize and package code.
	- Work on some other aspect such as testing experimental design.
  - [ ] Assess the results and if needed Improve the model and
    analysis. This will depend on whether we find an improved fit but
    the following is likely to be considered:
    - Diffusion - We are using a simplified space-discretised 2d
      model of diffusion
      - Use a finer grid?
      - Use a 3d grid?
    - Competition vs signalling
      - Are we correctly describing nutrient dependent growth and
    - Is the data in fact not dependent on competition or signalling?
      - Is growth actually independent?
      - Is there another factor which is introducing variance?
    - Does the analysis allow capture of the variance present in the
      data. c.f. (Addinall et al. 2011 &) Heydari et al. 2016 for an
      explanation.
  - [ ] Compare competition effects in different experimental
    designs in order to design experiments which minimze competition
    effects.
    - For instance, compare spot and agar geometries and initial nutrient
      densities.
    - It may help to model diffusion accross agar height using a
      barcode experimental desing because we then remain in 2d.
  - [ ] Develop a model of competition in terms of parameters of the
    logistic model of population growth in oder to compare fitness
    measurements to previous studies.
    - Adjust carying capacity (now time dependent) of the logistic
      growth model to depend on the net amount of nutrients available
      to each culture: ammount of nutrients already converted to dry
      mass plus remaining nutrients in the location. Ammount of
      nutrients starts at N0 and carrying capacity is normalized as
      such.
    - I have not thought about how to addapt the logistic growth model
      to account for singalling yet.
    - Fit this model to the raw timecourse data using the time course
      of nutrient fluxes found by first fitting the mass action
      kinetics model to the same data.
      - Can this be done simultaneosly/hierarchically, possibly all
        the way to genetic interaction measure?
  - [ ] Reanalyse telomere data using the modified logistic
    model in order to determine the effects on gene interaction.
  - [ ] Package the model in SBML. Conform to minimum information
    standards and publish in the biomodels database.
  - [ ] Document and package other analysis tools in python in order
    to distribute.

** Plan
   - This refers to the timescale of different parts of the project
     - Could create a Gantt chart

